### DISCUSSION TITLE
Path questions for -PATH parameter

### DISCUSSION THREAD

**Discussion by VeridianVoid** [darkoperator/Posh-SSH#585] [Q&A]
No logical version of Path works within windows to windows Posh-SSH. But same paths work in OS-level SCP.  Full example :

Via scp command, this works :
**scp the_user@the_host:"/C:/TEST_DIR/SOURCE_DIR/TEST_FILE.TXT" "C:\TEST_DIR\TARGET_DIR\newfile.txt"**

Setup for POSH : 
```
Remove-Item -Recurse -Force "C:\TEST_DIR\" | Out-Null
mkdir "C:\TEST_DIR\" | Out-Null
mkdir "C:\TEST_DIR\SOURCE_DIR" | Out-Null
mkdir "C:\TEST_DIR\TARGET_DIR" | Out-Null
dir > "C:\TEST_DIR\SOURCE_DIR\TEST_FILE.TXT" | Out-Null

$SFTP_Host        = "the_host"
$SFTP_UserName    = "the_user"
$SFTP_RawPassword = "my_pw!"

Import-Module Posh-SSH 

$SFTP_Password   = ConvertTo-SecureString $SFTP_RawPassword -AsPlainText -Force
$SFTP_Credential = New-Object System.Management.Automation.PSCredential ( $SFTP_UserName, $SFTP_Password )
$FTP_Session     = New-SFTPSession -Computername $SFTP_Host -Credential $SFTP_Credential -AcceptKey
```

Works great so far, but here is where it fails :
**Get-SFTPItem -SFTPSession $FTP_Session -Path "/C:/TEST_DIR/SOURCE_DIR/TEST_FILE.TXT" -Destination "C:\TEST_DIR\TARGET_DIR"**

Gives : "_The given path's format is not supported_."

The only format that does work is (this is the only way the command seems to work): 
**Get-SFTPItem -SFTPSession $FTP_Session -Path "../../TEST_DIR/SOURCE_DIR/TEST_FILE.TXT" -Destination "C:\TEST_DIR\TARGET_DIR"**

But as you can see, there's no way to specify the drive.

This is a trivial example, is there documentation for pathing with posh-ssh? Am I missing something? Is this a windows-ish bug?

**Comment by MVKozlov** [comment]
seems the problem is [here for files](https://github.com/darkoperator/Posh-SSH/blob/78db804aae9dc5d3981fbe410e74a5e2bd58d515/Source/PoshSSH/PoshSSH/GetSftpItem.cs#L180)
and [here for folders](https://github.com/darkoperator/Posh-SSH/blob/78db804aae9dc5d3981fbe410e74a5e2bd58d515/Source/PoshSSH/PoshSSH/GetSftpItem.cs#L159)

under v5.1 `[System.IO.FileInfo]::new('/C:/Users')` gives this error

I believe both should be changed to `[System.IO.Path]::GetFileName(remotepath)`

**Comment by darkoperator** [comment]
Gonna see when I can setup a windows server with OpenSSH to test. I also found this on the library side https://github.com/sshnet/SSH.NET/issues/171

**Comment by MVKozlov** [comment]
It does not work only on v5.1, v7 have no problems

**Reply by darkoperator** [reply]
ok, that helps, will try out the changes you recomended, just finished setting up a SFTP on Windows. Thinking of dropping 5.x and moving everything to 7.4 at some point, but in the mean time need to fix this

**Reply by MVKozlov** [reply]
5.1 still widely used because it intergrated
and afaik Azure use it too

**Reply by darkoperator** [reply]
Main reason is that SSH.NET is going all of it new cipher and algorithm support to .Net 8, if I move the module to .Net 8 it will only run on 7.4 and above, .Net 8 assemblies are not supported in 5.1 to 7.0-7.3

**Reply by MVKozlov** [reply]
I think it's really time to learn how to make a multiplatform module :)
Especially if the main difference is in the base library, and not in the module itself.
True, then the versions under different posh will have different features, but at least the baseline version support will remain

**Comment by darkoperator** [comment]
should be fixed now in 3.2.1, just pushed it to the gallery
